marker_rustc_driver/src/main.rs

```diff
@@ -170,14 +240,14 @@ fn main() {
             orig_args.extend(vec!["--sysroot".into(), sys_root]);
         };
 
-        // make "marker-driver --rustc" work like a subcommand that passes further args to "rustc"
-        // for example `marker-driver --rustc --version` will print the rustc version that marker-driver
-        // uses
+        // make "marker_rustc_driver --rustc" work like a subcommand that passes
```

Comment:
I see that previously dashes were used to separate package names. I don't understand why the choice was made to use underscores. To my opinion, crate and binary names should have kebab case. This is a very common convention. For example, cargo plugins are enforced to use kebab-case, tools like `rust-gdb`, `rust-gdbui`, `rust-lldb`, `typos-cli`. It's also confusing that you have to install `cargo_marker`, but then it is insalled as `cargo-marker`.

Reply:
Agreed, I can't tell you how many times I've typed the kebab case version when I need the underscore version, this seems relatively simple to change if we want to. @xFrednet might have a good reason for changing it though.

Reply:
:sweat_smile: Yeah, so, the background of this is that clippy uses underscores in their crates (`clippy_lints` & `clippy_utils`) for library crates. For library crates, this also makes sense IMO, since the `use` statement also uses underscores. I also looked at some other crates and there seemed to be no real consensus, which should be used bevy for example also always uses underscores.

Now for the binaries... when I decided on the name, I just reserved the `driver` and `cargo_marker` name without contemplating that these are binaries. Having underscores everywhere just seemed consistent. That was until I found out that Cargo expects the binary to use a dash instead of an underscore.

So, this basically comes down to, little experience when it comes to actually releasing crates/binaries and only learning about my mistakes after it was too late. I even emailed crates.io about maybe changing the name, but they said that this is not possible.

We could rename the binary in the same way that the `cargo_marker` crate is renamed, but the names are sadly stuck now. (The only other option, I see, would be to rename the project, which I don't want to)

Reply:
I see, that's unfortunate, but I don't see a better solution than renaming

Reply:
There is one solution for the binaries: Make Marker successful enough that it's actually distributed with rustup instead ^^